patches/yaml-language-server+1.11.0.patch

```diff
@@ -0,0 +1,16 @@
+diff --git a/node_modules/yaml-language-server/.DS_Store b/node_modules/yaml-language-server/.DS_Store
+new file mode 100644
+index 0000000..151aa34
+Binary files /dev/null and b/node_modules/yaml-language-server/.DS_Store differ
+diff --git a/node_modules/yaml-language-server/out/server/src/commands.js b/node_modules/yaml-language-server/out/server/src/commands.js
+index 3f1643e..74d3a6c 100644
+--- a/node_modules/yaml-language-server/out/server/src/commands.js
++++ b/node_modules/yaml-language-server/out/server/src/commands.js
+@@ -7,6 +7,6 @@ Object.defineProperty(exports, "__esModule", { value: true });
+ exports.YamlCommands = void 0;
+ var YamlCommands;
+ (function (YamlCommands) {
+-    YamlCommands["JUMP_TO_SCHEMA"] = "jumpToSchema";
++    YamlCommands["JUMP_TO_SCHEMA"] = "jumpToArchitectIoSchema";
+ })(YamlCommands = exports.YamlCommands || (exports.YamlCommands = {}));
+ //# sourceMappingURL=commands.js.map
```

Comment:
This seems pretty sketchy and I don't think it's actually solving anything other than making an error go away - from what I can tell in my testing:
![image](https://user-images.githubusercontent.com/1406391/212399960-92678a57-31f8-4e80-b925-d3a4a62b464a.png)

This `architect.yml - Architect.io Component Schema` string is coming from SchemaStore here: https://github.com/SchemaStore/schemastore/blob/master/src/api/json/catalog.json#L232 and if you disable the RedHat YAML extension, the link itself no longer works and the schema doesn't open. This means that if somebody only installs our extension, this functionality doesn't work, so patching this value to change it seems pointless.

We can't get away with just patching it - I'm not sure what other options we have, but here's one idea:
- Because both this and the YAML extension use the same yaml server, ideally if the yaml language server is already provided by another extension, we could disable also configuring it here and rely on that installation. There's some risk to this where if our extension is out of sync with the RH yaml plugin, it's possible there's breaking changes. I also don't know if what I said is possible, I don't exactly know how the yaml language server gets set up, but that should be looked into/understood so we know our options.

Reply:
Ah yes, that is a side effect for collision avoidance between vscode-yaml and other apps using yaml-language-server. This patch's purpose is purely to remove the collision of vs-code command registration. Without the patch, whichever conflicting extension that opens second will try to re-register a `JumpToSchema` command and fail to start entirely once the conflicting language is selected (in this case architectio -> yaml , and yaml -> architectio).

Conditionally registering the command without the patch would only work if someone switches from YAML language to architectio language. But not the other way around.
eg.  
architectio installed, architect.yml file is opened and language server is activated and jumptoSchema is registered.
vscode-yaml is then installed and a .gitlab-ci.yml file is opened.  it's going to register jumptoSchema regardless, but it's already been registered this session, so vscode-yaml fails to launch completely.

There doesn't seem to be an unregister command that I could find for extensions. We provide links to the schema from the hover box and on the marketplace from our readme. Downloading the schema file directly doesn't seem to be a requirement for MVP. So I still think it is a good solution for the MVP until more discovery can be made for:
https://github.com/redhat-developer/yaml-language-server

PS: eventually writing our own language server would be ideal, however would be pretty far out of scope for the reqs for v1

Thoughts?

Reply:
I definitely don't think we should consider writing our own language server any time soon - but as it stands this patch does nothing except prevent an error from showing up and is super brittle if the underlying language server changes at all. Even if the feature to click and jump to schema is important, I don't think we want to settle for doing something this hacky to work around forking the repo...

From some GH discussion on this issue: https://github.com/redhat-developer/yaml-language-server/issues/765 It seems like the recommended way of using `yaml-language-server` and `vscode-yaml` is to actually just have `vscode-yaml` as an extension and add our specific bits on top of it. This might be our best bet going forward - as it stands now, we have a somewhat modified somewhat stripped down fork of `vscode-yaml` that is impossible to keep up to date with any changes they make (because we can't just pull in their changes, so any bugs that exist in their code we are kind of stuck with or need to manually mirror the fix into our repo). If we can instead write some code on top of `vscode-yaml` that just modifies it's behavior (to e.g. remove the functionality we don't care about like allowing more than one schema and such), 

It se ems like we can declare `extensionDependencies` in the package.json: https://code.visualstudio.com/api/references/extension-manifest
```
extensionDependencies |   | array | An array with the ids of extensions that this extension depends on. The id of an extension is always ${publisher}.${name}. For example: vscode.csharp.
```

which would presumably allow us to use the yaml extension as a dependency and share `yaml-language-server`, and also get us out of the business of fixing any bugs within the `vscode-yaml` extension itself

Reply:
I searched around for an open source example of this just to see if it's been done before and found this:

https://github.com/ansible/vscode-ansible
extensionDependencies here - https://github.com/ansible/vscode-ansible/blob/main/package.json#L549

This does obviously a lot more than we need to be doing on top of the `vscode-yaml` extension but is an example of building on top of `vscode-yaml` rather than reworking it

Reply:
Update here - I spent way too long on Friday night seeing if there was a way to create a client that connected to an existing language server installed by another extension, but this seems to not be possible (and if it is possible somehow, it'd be incredibly incredibly hacky).

I chatted with TJ & Michael to make sure we're all on the same page, and the best path forward is to do the following:

- instead of patching this dependency, we should fork the `yaml-language-server` repo and modify it to change the name of this command
- This extension should use our forked language server
  - This sets us up in the future to make modifications to the language server itself, which would be needed if we added things like interpolation support, etc.
  - If there's any other bugs that we just haven't seen yet related to running an identical language server, we can more easily fix them
  - We can easily keep our fork up to date with the upstream repo and don't have to worry about the patch breaking if code moves around
- We should test if the "Jump to schema" functionality still isn't working and figure out why - we don't want something to only work if you have a different extension. I'm happy to take a look into this myself

Reply:
* Removed patch and patch dependencies. 
* Removed yaml-langauge-server and dependencies
* Forked architectio-language-server
* Modified architectio-language-server yamlHover, yamlCommands, and yamlHover
* updated jumpToSchema enum value to jumpToArchitectioSchema
* pushed extension disposable for workspace using registerTextDocumentContentProvider for architectio schema type with new class ArchitectioSchemaDocumentContentProvider to call getSchemaContent when attempting to load the schema in a new window when codelens action is registered
* correctly loads schema with or without vscode-yaml